File: middlerApp.API/IDP/DtoModels/MRoleDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using middlerApp.API.IDP.Models;

namespace middlerApp.API.IDP.DtoModels
{
    public class MRoleDto
    {
        public Guid Id { get; set; }

        public string Name { get; set; }
        public string DisplayName { get; set; }
        public string Description { get; set; }

        public ICollection<MUserListDto> Users { get; set; } = new List<MUserListDto>();
    }
}

File: middlerAPP/src/app/areas/admin/components/idp/models/m-role-dto.ts
export class MRoleDto {
    Id: string;
    Name: string;
    DisplayName: string;
    Description: string;


    /// Helper Properties
    Deleted: boolean = null
}

export function MRoleDtoSortByName(a: MRoleDto, b: MRoleDto) {
    if (a.Name?.toLowerCase() < b.Name?.toLowerCase()) {
        return -1;
    }
    if (a.Name?.toLowerCase() > b.Name?.toLowerCase()) {
        return 1;
    }
    return 0;
}